cylc/flow/run_modes/nonlive.py

```diff
@@ -0,0 +1,55 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilities supporting all nonlive modes
+"""
+from typing import TYPE_CHECKING, Dict, List
+
+from cylc.flow import LOG
+from cylc.flow.run_modes.skip import check_task_skip_config
+from cylc.flow.run_modes import RunMode
+
+if TYPE_CHECKING:
+    from cylc.flow.taskdef import TaskDef
+
+
+def run_mode_validate_checks(taskdefs: 'Dict[str, TaskDef]') -> None:
+    """Warn user if any tasks have "run mode" set to skip.
+    """
+    warn_nonlive: Dict[str, List[str]] = {
+        RunMode.SKIP.value: [],
+    }
+
+    # Run through taskdefs looking for those with nonlive modes
+    for taskdef in taskdefs.values():
+        # Add to list of tasks to be run in non-live modes:
+        if (
+            taskdef.rtconfig.get('run mode', None)
+            in {
+                RunMode.SIMULATION.value,
+                RunMode.SKIP.value,
+                RunMode.DUMMY.value
+            }
+        ):
+            warn_nonlive[taskdef.rtconfig['run mode']].append(taskdef.name)
+
+        # Run any mode specific validation checks:
+        check_task_skip_config(taskdef)
+
+    if any(warn_nonlive.values()):
+        message = 'The following tasks are set to run in skip mode:'
+        for taskname in warn_nonlive[RunMode.SKIP.value]:
+            message += f'\n    * {taskname}'
+        LOG.warning(message)
```

Comment:
What is the rationale for emitting this warning?

Reply:
To ensure that users don't unwittingly end up configuring tasks to skip in dev then running for real.

See proposal.

Reply:
What about this section in the docs PR - it suggests a use case for using a skip mode task in a production workflow: https://github.com/cylc/cylc-doc/pull/755/files#diff-59a37832de8ec0a5bf3eabdb98220f5e6ef55a14d61abe6810c4d4b43160345fR151-R181

Reply:
Then the user will get a warning. I'm not sure that we can reverse engineer the motives for running a task in skip mode, so we haven't got enough information.

The only way I can think of doing it is 

```
if not task.name.startswith("skip"):
  warn
```

It may require extra docmentation from workflow writers (You can ignore the warning)